### ISSUE TITLE
seed not sufficient for exact reproduction

### ISSUE THREAD

**Issue by jowagner** [CoNLL-UD-2018/UDPipe-Future#14]
For 36 of my models, I ran training twice with the same .conllu input files and same seed but never obtained the same results in the two runs. The logs shows tiny 0.01 differences in some of the evaluation metrics in early epochs, which then grow and more metrics start showing differences. At the end of training (default settings), differences of 0.10 to 0.20 (on the 0-100 scale) are frequent. I've two types of GPUs on the cluster (rtx2080-ti and tesla K40) but it is unlikely that the type of GPU was different in all 36 cases, so I don't think it's the gpu type. I remember reading years ago that GPUs calculate less accurately than CPUs. Maybe it's that.

It may be a good idea to mention in documentation for the `--seed` option that it cannot guarantee reproduction of results due to unpredictable inaccuracies of calculation on GPUs.

**Comment by foxik** [comment]
This is a known TensorFlow feature -- many GPU kernels are stochastic (because of scheduling of multiple threads running the same kernel and many others), see for example https://github.com/tensorflow/tensorflow/issues/2732 and https://github.com/NVIDIA/tensorflow-determinism.
